app/controllers/vacancies_controller.rb

```diff
@@ -48,6 +59,24 @@ def set_headers
     response.set_header("X-Robots-Tag", "noarchive")
   end
 
+  def campaign_search_params
+    params.permit(:email_name, :email_postcode, :email_location, :email_radius, :email_jobrole, :email_subject,
+                  :email_phase, :email_ECT, :email_fulltime, :email_parttime, :email_jobshare, :email_contact)
+          .tap do |campaign_params|
+      campaign_params[:location] = campaign_params.delete(:email_location)
+      campaign_params[:radius] = campaign_params.delete(:email_radius)
+      campaign_params[:teaching_job_roles] = [campaign_params.delete(:email_jobrole)].compact
+      campaign_params[:subjects] = [campaign_params.delete(:email_subject)].compact
+      campaign_params[:phases] = [campaign_params.delete(:email_phase)].compact
+      working_patterns = []
+      working_patterns << "full_time" if campaign_params.delete(:email_fulltime) == "true"
+      working_patterns << "part_time" if campaign_params.delete(:email_parttime) == "true"
+      working_patterns << "job_share" if campaign_params.delete(:email_jobshare) == "true"
```

Comment:
should we not use ActiveModel::Type::Boolean.new.cast(value) here so that we cope with values of "1" (and Yes?) as well as (lowercase) true?